components/admin/lessons/AdminLessonInputs/AdminLessonInputs.tsx

```diff
@@ -60,48 +68,42 @@ const initValues = (
   }
 ]
 
-const AdminModuleInputs = ({
+const AdminLessonInputs = <MainItem extends Item>({
   title,
   lessonId,
-  onAddModule,
-  module,
-  refetch
-}: Props) => {
+  item,
+  loading,
+  refetch,
+  onActionFinish,
+  action
+}: Props<MainItem>) => {
   const [formOptions, setFormOptions] = useState<any>(initValues())
   const [name, content, order] = formOptions
 
+  const [data, setData] = useState<null | undefined | typeof item>(null)
+
   useEffect(
     () =>
       setFormOptions(
-        initValues(
-          get(module, 'name'),
-          get(module, 'content'),
-          get(module, 'order')
-        )
+        initValues(get(item, 'name'), get(item, 'content'), get(item, 'order'))
       ),
-    [module]
+    [item]
   )
 
   const mutationVariables = {
-    content: content.value,
+    content: content.value as string,
```

Comment:
Remove `as string`

Reply:
I added it because by default its type is `any`, and it's needed (for the sake of keeping things type-safe) because 
`formOptions` type is `any` in `useState<any>(initValues())`.